nautobot_ssot/mixins.py

```diff
@@ -0,0 +1,175 @@
+"""Mixins to be used when working with Nautobot."""
+
+from typing import Optional
+from diffsync.enum import DiffSyncFlags, DiffSyncModelFlags
+
+
+def get_fk(fks, diffsync, key, value):
+    """Function to get the fk of an object, given information stored in ORM."""
+    # fks comes from self._foreign_keys
+    # key matches a key in `_foreign_keys` which is the local attribute
+    # value is the get_unique_id() we are looking for
+    if key in list(fks.keys()):
+        model = [val for _key, val in fks.items() if _key == key][0]
+        key_model = diffsync.meta[model]
+        pkey = diffsync.get(model, value).pk
+    return key_model.objects.get(pk=pkey)
+
+
+class DiffSyncModelMixIn:
+    """MixIn class to handle sync generically."""
+
+    _foreign_key = {}
+    _many_to_many = {}
+    _generic_relation = {}
+    _unique_fields = ("pk",)
+    _skip = []
+
+    pk: Optional[str]
+
+    @classmethod
+    def create(cls, diffsync, ids, attrs):
+        """Create Method to handle generically."""
+        model = super().create(ids=ids, diffsync=diffsync, attrs=attrs)
+        # model is the DiffSyncModel instance, which has data about the instance required
+        db_model = cls.Meta.model
+        obj = db_model()
+        instance_values = {**ids, **attrs}
+        combined = (
```

Comment:
`combined` what? maybe better `combined_refernce_keys`? Trying to get a name that better describes the intention and helps to understand